sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/ext/FileExt.kt

```diff
@@ -17,3 +18,22 @@ internal suspend fun File.canWriteSuspend() = withContext(Dispatchers.IO) { canW
 internal suspend fun File.renameToSuspend(dest: File) = withContext(Dispatchers.IO) { renameTo(dest) }
 
 suspend fun File.deleteRecursivelySuspend() = withContext(Dispatchers.IO) { deleteRecursively() }
+
+/**
+ * Encrypts File to SHA1 format. This method is not recommended on huge files. It has an internal limitation of 2 GB
+ * byte array size.
+ *
+ * @return String SHA1 encryption of the input file
+ *
+ * @throws OutOfMemoryError if the file is too big to fit in memory.
+ */
+@Throws(OutOfMemoryError::class)
+fun File.getSha1Hash(): String {
+    return MessageDigest
+        .getInstance("SHA-1")
+        .digest(readBytes())
```

Comment:
This will consume ~50mb of ram, which is a lot.

I think we can do a more efficient implementation with https://developer.android.com/reference/java/security/DigestInputStream which streams the file input.

Reply:
Yes, good point. I updated the implementation to work with streams.